types/chai/index.d.ts

```diff
@@ -725,7 +724,7 @@ declare namespace Chai {
          * @param value   Actual value.
          * @param message   Message to display on error.
          */
-        isArray<T>(value: T, message?: string): void;
+        isArray<T>(value: T, message?: string): asserts value is (T extends any[] ? T : never);
```

Comment:
I don't think this type is quite right. A narrower type than `any[]` could be passed in (like `{}`) and still pass the assertion.